### PR TITLE
Permagunk

### PR DESCRIPTION
**Pull request by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE**
Blood, gibs, footsteps, general filth, and trash will now all persist through a maximum of 5 rounds if not cleaned.
Only up to 5 per tile are allowed, to prevent massive gunkpiles.
This will finally give the Janitor a purpose.

Blood and gibs become noticeably darker over rounds, so detectives don't try to scan them and wonder why it doesn't work.
dreamseeker_2019-02-10_21-17-40
dreamseeker_2019-02-10_21-19-30
dreamseeker_2019-02-10_21-22-14

Non-playermade trash and gunk that was part of the map will not persist, to prevent gunk feedback.
Soot, polyacid goo, and fuel will not persist for reasons.

dreamseeker_2019-02-10_03-32-24
NO lag

Does not work for shuttles. Does not work for Centcomm. Does not work for vaults. Does not work for away missions. Does not work in space. Does not work inside walls (if you remove a wall and put trash there, it will be telefragged by the wall next round)
Does work for the Derelict, asteroid, etc.
I tried mass drivering 200 pieces of trash and the only ones that managed to land in a persistent-able spot afterwards were all in the Vox ship docking area. Which is totally a fucking feature because now you're not just cleaning up, but trans-spacially littering the Vox as well.
dreamseeker_2019-02-08_05-14-21

Comes with an admin-panel to view persistence stats, and big DELETE ALL buttons Should Anything Go Wrong.
Nar-Sie, nukes, and supermatter sea all disable persistence so the next round will be squeaky clean.

Much of this was copied from https://github.com/Baystation12/Baystation12/pull/22928/, mad credits to MistakeNot4892

Complimentary video: 
https://youtu.be/gAXsLOF05TY?t=158

:cl:
 * experimental: The following changes are a test merge. They will be reverted in one week, then a poll about them will be held.
 * experimental: Blood and trash will now persist for up to 5 rounds if not cleaned (only up to 5 per tile). Persistent blood doesn't have DNA, carry viruses, work for cult objectives, etc. the only thing it does is give janitors something to do. Have fun!
 * experimental: Persistent blood will be named 'dry blood' and look noticeably darker so it doesn't throw detectives off.
 * experimental: For every 100 persistent filth entries, an extra cleaner grenade will spawn in the Janitor's closet in all maps.
 * experimental: Nukes, supermatter sea, and Nar-Sie will disable filth persistence and make the next round clean again.

### CODE REVIEW

**Comment by SonixApache** [comment]
wew

**Comment by DrSnips** [comment]
I'm not sure I like this since each round is supposed to be individual and doesn't/shoudn't have persisting effects. We have rounds where the universe literally ends and we don't see that show up at all. It's a cute feature but it just doesn't sit well with me.

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
I partially agree. That's why I forbid crayon writing and any form of text from being persistent.
Filth only, I think, should be generic enough to uphold muh disbeliefz

**Comment by Allspice** [comment]
noice

**Comment by Intigracy** [comment]
what's `comingdir_goingdir_and_bloodcolor_in_that_order` used for? the var name isn't really helpful

:^)

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
I had it written down somewhere but I lost it

**Comment by DrSnips** [comment]
This still doesn't feel right. The station should start nearly pristine outside of maint. Having blood and gibs everywhere for several rounds sounds wrong.

**Comment by ShiftyRail** [comment]
That's a terrible idea
The station should come to a gunk state naturally, not start in it

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
I don't see why the station "has" to start pristine. In fact there's trash already mapped in, even outside of maint, especially in non-box maps.

**Comment by Goldsteel** [comment]
I'm not sure how I feel about this. This kind of reinforces the crew transfer ideas, but often the station is utterly and hideously destroyed and left in a mess. Why would the station regenerate but the mess would be left uncleaned?

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
Basically, to make trash matter more and give the Janitor stuff to do. Why bother cleaning if it'll be gone next round?

The time paradox thingy is a side-effect if anything. If we're all grown vegetables that can pretend to ignore deadchat after being cloned, then we can pretend to not know how these gibs got there.

**Comment by codeungabunga** [comment]
The black blood looks weird.

**Comment by Goldsteel** [comment]
Maybe cut it down from 5 rounds to 2-3. Some rounds are going to be horrendous and often there won't be any janitorial staff. We have a lot of means to clean things up like space cleaner, soap, mopping, the janicart, smoking space cleaner and bleach, wiping things down with wet rags.

My biggest worry is that people will make tons of mess on purpose in weird places just to bother people between rounds, especially with bloodbath rounds like cult that will leave the next station in a complete disaster state. There's much to be said about the psychology behind "If it looks like shit, treat it like shit".

**Comment by ShiftyRail** [comment]
I can already imagine the result of a very long, very disorderly and very messy lowpop round 
It's not rare that these rounds end with every single tile covered in blood or another fluid

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
@codeungabunga I agree, I toned down the darkness of dried blood somewhat, check out the updated screenshots

File: code/game/objects/effects/decals/cleanable.dm [hunk 1]
```diff
@@ -17,11 +17,66 @@
 
 	var/on_wall = 0 //Wall on which this decal is placed on
 
-/obj/effect/decal/cleanable/New()
-	if(random_icon_states && length(src.random_icon_states) > 0)
+	var/persistence_type = SS_CLEANABLE
+	var/age = 1 //For map persistence. +1 per round that this item has survived. After a certain amount, it will not carry on to the next round anymore.
+	var/persistent_type_replacement //If defined, the persistent item generated from this will be of this type rather than our own.
+	/obj/item/trash/New(var/loc, var/age, var/icon_state, var/color, var/dir, var/pixel_x, var/pixel_y)
```
Comment (DamianX) [inline code/game/objects/effects/decals/cleanable.dm]: the indentation is wrong

Comment (DamianX) [reply code/game/objects/effects/decals/cleanable.dm]: oh actually it's a duplicate definition, trash.dm:14

**Comment by bathosbathtime** [comment]
Very long, very messy rounds should be impactful.  I don't have know if persistence through rounds is a good idea but I find this to possibly be the tamest way to test if it is something that would fit on our server.

File: code/game/objects/effects/decals/cleanable.dm (hunk 1 above)
Comment (I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE) [reply code/game/objects/effects/decals/cleanable.dm]: How did this even compile?

**Comment by codeungabunga** [comment]
That looks much more like blood.

If people are concerned about massive amounts of trash, it could have a cap for max trash and cap for max decals, and just pick at random from those that remain after round ends until it's filled, overwriting old ones.

**Comment by SonixApache** [comment]
Does ALL of it persist? That seems like kind of a lot.

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
A cap can easily be added if people find it too much. Or better, add a cleaner grenade spawned in the Janitor's office every 100 persistent filth tiles.

**Comment by DeityLink** [comment]
Whether it turns out to be a good or a bad idea, I NEED to see it in action. This concept makes me want to pick up janitor again.

THAT BEING SAID. @I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE Can you had an exception for cult rounds that ended in cult victory? since not only would the Station canonically no longer be part of the rest of the universe, but given the gigantic amount of blood-painted tiles needed to achieve cult victory, that might prove too much work for the Janitor, and also ruin the mood a bit for the rest of the crew.

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
Yeah I'll get on that.

**Comment by Jeroen52** [comment]
@DeityLink Probably better for any station-ending event.

**Comment by Blithering** [comment]
Multi-round viroloose?

**Comment by Irradiation** [comment]
I know the spread blood jectie is already a meme jectie because sometimes the station is bloody enough pre-sacrifice, but good lord, now it's going to be 100% always a meme.

**Comment by DeityLink** [comment]
I can easily up a bit the amount of bloody tile needed to make up for that.

**Comment by Blithering** [comment]
or make the objective be number of bloody tiles at time of objective activation + n

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
Dried blood doesn't count for gult

**Comment by SonixApache** [comment]
👎 
I for one can't wait for the playerbase to make it their sole objective to cover every tile on the station in dust, goo, blood, blood, more blood and vomit every round, alongside cult auto-skipping a stage.

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
@SonixApache dried blood doesn't count for gult

Also, spreading filth is technically against spacelaw but nobody cared until now, I'm excited to see the evolution of that law in regards of dealing with it IC

**Comment by yclatious** [comment]
You people will learn the meaning of regret once this shit is merged and I get to play.

Filthnades will finaly shine in all their hideous, disgusting nature. Mark my words.

**Comment by kilozombie** [comment]
Fuck it, let's do it.

If things get horrendously bad, it's fixable. I want to experience it if that's the case, and I think if it ends up being manageable, it'll be fun as hell.

**Comment by Accipitermaris** [comment]
Abstract: If merged, I think this PR going to have a lot of unintended consequences Broken Windows Theory style. I'm by no means a social psychologist though.

Specific: **If merged, I think it's crucial that we ensure that all departments have reasonable basic cleaning supplies!** This is to ensure that an oppressive "Cargo Roundstart Routine" isn't hammered in for Engineers and Civilians on top the "Departmental Round Progression" already in effect.

-Security, Service, Medical, and Cargo all have reasonable supplies/means to keep their spaces clean, however Civilians(!) and Engineers(!) are quite limited in this regard save for Cargo Routines. (I'm unsure about Science as I haven't really played that job.) A simple mapping of more departmental soaps would suffice here, as well as a few random soap closets in maint, or, if that is too liberal, soap in offices of the Civilian jobs that maintain a work space (Librarian, Chaplain, IAA.)

-I also think that Space Cleaner Bottles should be constructable with Plastic (Or, alternatively, with plastic loaded into an Autolathe if people don't want sprayer bottles to be too easy to obtain(?).)

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
- Nukes, supermatter sea, and Nar-Sie now disable filth permanence and make the next round start squeaky-clean. Admins can also force this if they want to.
 - For every 100 filth entries on the station, an extra cleaner grenade will spawn in the Janitor's closet, to counter mr. edge over here. This may be too generous, but we'll wait and see

@Blithering Dry blood carries no viruses

**Comment by Jeroen52** [comment]
@I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE Any tiles cleaned by Mr. Clean should be clean, in case they're not.

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
Any method of cleaning tiles removes them from the gunk savefile

**Comment by SonixApache** [comment]
more spray cleaners for people to slipp with, great

**Comment by jknpj** [comment]
IMAGINE bloodnades

**Comment by yclatious** [comment]
>fearing bloodnades
>only bloodnades

My man, you aint seen shit, ill make Viscera Cleanup look like a fucking joke.

File: code/game/objects/effects/decals/cleanable.dm (hunk 1 above)
Comment (ShiftyRail) [reply code/game/objects/effects/decals/cleanable.dm]: http://www.byond.com/forum/?post=2408847

It's a feature apparently

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
For as long as the soap has no attack cooldown, stacking multiple types of decals on top of eachother doesn't do anything really except spawn more free cleanernades since blood+radium+carbon+gibs is just as easy to clean as just blood

**Comment by Kuranon** [comment]
Even 10% of trash transferring over seems like too much, and you seriously want ALL of it to transfer? There needs to be more limits than just "only up to 5 per tile"

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
I'm not opposed to the idea of adding a limit to it, but I think this probably sounds worse than it really is. I'll be paying attention the first couple of days to see what people think

**Comment by DeityLink** [comment]
That's the spirit. A brand new feature is bound to be imbalanced and require a bit of tweaking. And I'll be sure to experience it in its raw glory first.

**Comment by Durgen** [comment]
I prefer the status quo, as a clean station at roundstart creates implicit lore that the station you are deployed on is very boring and ordinary. This meshes with server rules such as not powergaming too much, being chill on code green, etc.

If rounds begin with Central Primary looking like a slaughterhouse, it will be harder to enforce these rules/mindset.

Could we maybe restrict persistent messes to maint as a compromise?

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
But then we couldn't have "ugh, the previous chemist splashed radium on the floor" scenarios

If a limit has to be added, the first thing I'll add is a limit of filth tiles per room. Again though, I think this sounds worse than it really is, so we'll see

**Comment by spoonylove** [comment]
>crew transfer into new station
>take one step off arrival shuttle
>As a fuck you the clown from last shift lubed up all the airlocks of the shuttle
>slip the instant I get onto station

WOW. Not possible but would be funny. Giving the janitor more stuff to do sounds like a good change, would make things like certain maps having multiple janitors actually mean something now that they have a goal. Should really test out having persistent graffiti and drawings so that assistants can tag territory for their maint gangs for multi round turf war memes.

**Comment by Kitfox88** [comment]
not seeing how this adds anything fun or interesting other than just giving an incentive for assholes to make messes all the time even more than they do now

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
Cleaning up, obviously, and assholes making a mess will be treated more severely

**Comment by Exxion** [comment]
This is a fucking terrible idea
But I guess the revert button exists for a reason

**Comment by Irradiation** [comment]
Incidentally, doesn't this PR make the previous rounds canon because there's physical proof of the events that occured in the form of trash, bloods and gibs?

God damnit, this PR ruined my "this is all a simulation" headcanon.

**Comment by Exxion** [comment]
Good because that headcanon is almost "it was all a dream"-tier
But also not really, since it doesn't keep the big fucking holes in the station

**Comment by DeityLink** [comment]
I prefer my headcanon of "multiple stations" or "parrallel universes"

**Comment by SonixApache** [comment]
testmerge when

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
I'd say put it up for a week, revert it, then make a poll

**Comment by Gringonius** [comment]
Why?

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
Gold standard for test merges

**Comment by Exxion** [comment]
I know part of the idea of this is to give the janitor a purpose, but keep in mind the janitor doesn't have all access. So departments definitely need their own cleaning supplies with this.

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
All of them except Engineering already have soap, the greatest cleaning tool in the game
How does moving the soap from mechanics to engineering lobby sound? Why do mechanics even have soap anyways?

**Comment by DeityLink** [comment]
I want my departmental janitorial closets!

**Comment by Exxion** [comment]
Some of the pseudo-departments like library and chapel have offices that the janitor can't get to and don't have soap, but they're pretty small and don't usually get messy so I guess it's not a huge deal
Also, all of them? I'm specifically not sure about sec, cargo, or the bridge (aside from the captain's soap, which seems weird but I guess works)

**Comment by kilozombie** [comment]
With personal offices and the like, I think it's fair to even just radio "Hey, janitor needed here"
The new HoP console means we can have more janitors, too.

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
To be honest the paramedic and mechanic are other examples of people regularly expected to go into other people's offices and don't have all access either

**Comment by Exxion** [comment]
Which is kind of why paramedic is considered rather shit and mechanics often don't do that part of their job

**Comment by IzayaChan** [comment]
If this doesn't merge because of muh lore I will domp eet

**Comment by Exxion** [comment]
If it doesn't get merged it won't be because of muh lore
But it most likely will get merged thanks to the mostly positive votes

**Comment by Durgen** [comment]
reminder of a possible compromise I mentioned earlier: perma gunk in maint only

I think this would mostly assuage peoples lore/head cannon concerns, while also giving janitors something to do

inb4 wyci

**Comment by Gringonius** [comment]
'Why?' As in "why have this feature at all?"

**Comment by yclatious** [comment]
Why do we have a thermonuclear reactor solely to make gas hot and shove it in a turbine generator? Why does a research station have so many departments and such a well stocked armory?  Why do we keep employing clowns, vampires, traitors and incompetent comdoms?

Why have bannanons, supermatter crystals, space lube or anything else? Why do we have wizards who xan litteraly buttblast your booty away, have gods appear into reality to shitpost and have unironic cults to the Honkmother and its pussy version, Nar Sie?

For fucking fun, thats why.

**Comment by bathosbathtime** [comment]
It gives janitors reason to be let into messy departments, it tests out persistence on the server, it is a neat idea with a more positive reaction than negative, and because he coded it.

**Comment by Gringonius** [comment]
How is a bloated amount of trash and dirt 'fun'?
Make this exclusive to deff, at least.

**Comment by bathosbathtime** [comment]
It gives a stronger impact to making a mess so our users that enjoy that will have more fun.  Simultaneously it gives the janitor reason to enter other departments and clean up, even as the round draws to a close, so the players that enjoy making sure everything is neat and tidy will be taken more seriously.

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
I'll do the "remove soap from mechanic add it to engineering lobby" thing tomorrow and maybe add a second soap in the warden's office so they don't have to get it from perma.

**Comment by Exxion** [comment]
Cargo proves regularly that they can open anything by themselves

**Comment by Accipitermaris** [comment]
>Also, if Cargo orders cleaning supplies can they open them by themselves?
Yeah, janitorial cargo crates are not ID-locked.

**Comment by Intigracy** [comment]
fix conflicts

**Comment by Kyunny** [comment]
I like the idea overall.
If cleaning spaces janny has no access to (like engine rooms) proves too much of a trouble then maybe disable the feature inside some places.

**Comment by Kyunny** [comment]
Or making only 30% of the filth persist.

**Comment by Kyunny** [comment]
Or letting admins turn the feature on and off in-game.

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
You see, when I said "I'll do it tomorrow" I obviously meant "I'll wait until the roidstation PRs are merged because they will conflict every damn time" sure or course

I'm busy this week sorry

**Comment by Intigracy** [comment]
fix conflicts

**Comment by Kurfursten** [comment]
Update on this?

**Comment by Jeroen52** [comment]
fix pls

**Comment by Intigracy** [comment]
I want to merge this but you're not making it easy.

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
Closing until I have the time to fix this + deal with the changes people will request after the test merge

**Comment by Jeroen52** [comment]
\*screams autistically\*

**Comment by SirSkeleto** [comment]
merge soon
emojiocracy demands it

**Comment by Intigracy** [comment]
Now I'm sad.

**Pull request closed without merging**